app/controllers/email_authentications/invitations_controller.rb

```diff
@@ -1,29 +1,51 @@
 class EmailAuthentications::InvitationsController < Devise::InvitationsController
-  before_action :verify_params, only: [:create]
+  before_action :verify_params, only: [:create], unless: -> { Flipper.enabled?(:new_permissions_system_aug_2020, current_admin) }
+  before_action :🆕verify_params, only: [:create], if: -> { Flipper.enabled?(:new_permissions_system_aug_2020, current_admin) }
   helper_method :current_admin
 
   def new
-    authorize([:manage, :admin, current_admin])
+    if Flipper.enabled?(:new_permissions_system_aug_2020, current_admin)
+      raise UserAccess::NotAuthorizedError unless current_admin.can?(:manage, :facility)
+    else
+      authorize([:manage, :admin, current_admin])
+    end
+
     super
   end
 
   def create
-    user = User.new(user_params)
-    authorize([:manage, :admin, user])
+    if Flipper.enabled?(:new_permissions_system_aug_2020, current_admin)
```

Comment:
Why do we key off the flag here? I thought that the `create` method will continue to do its thing, and we'll use Flipper to redirect darklaunched users to the `new_create` path instead?

Reply:
I removed the `new_create` method, it didn't buy us much. Instead it spills awkward devise code all over the routes files and paths in views. It'll be easier to remove this code later if it's all in the same method, so I just kept it here.